NewsApp-main/NewsApp/Scenes/NewsDetail/ViewController/NewsDetailViewController.swift

```diff
@@ -0,0 +1,370 @@
+        if collectionView == sliderCollectionView {
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: NewsDetailCollectionViewCell.identifier, for: indexPath) as? NewsDetailCollectionViewCell else {
+                return UICollectionViewCell()
+            }
+            cell.setNewsDetail(viewModel.topNews[indexPath.row])
+            cell.saveButton.tag = indexPath.row
+            cell.saveButton.addTarget(self, action: #selector(saveSlideItems(_:)), for: .touchUpInside)
+            if let data = viewModel.fetchData()?.filter({ $0.title == viewModel.topNews[indexPath.row].title }) {
+                if !data.isEmpty {
+                    cell.saveButton.setTitle("Okuma Listesinden Çıkar", for: .normal)
+                    cell.saveButton.setTitleColor(.black, for: .normal)
+                    cell.saveButton.titleLabel?.font = UIFont.systemFont(ofSize: 12)
+                } else {
+                    cell.saveButton.setTitle("Okuma Listesine Ekle", for: .normal)
+                    cell.saveButton.setTitleColor(.gray, for: .normal)
+                    cell.saveButton.titleLabel?.font = UIFont.systemFont(ofSize: 12)
+                }
+
+            }
+            return cell
+        } else if collectionView == newsListCollectionView {
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: NewsDetailCollectionViewCell.identifier, for: indexPath) as? NewsDetailCollectionViewCell else {
+                return UICollectionViewCell()
+            }
+            cell.setNewsDetail(viewModel.sourceDetailList[indexPath.row])
+            cell.saveButton.tag = indexPath.row
+            cell.saveButton.addTarget(self, action: #selector(saveNewsItems(_:)), for: .touchUpInside)
+            if let data = viewModel.fetchData()?.filter({ $0.title == viewModel.sourceDetailList[indexPath.row].title }) {
+                if !data.isEmpty {
+                    cell.saveButton.setTitle("Okuma Listesinden Çıkar", for: .normal)
+                    cell.saveButton.setTitleColor(.black, for: .normal)
+                    cell.saveButton.titleLabel?.font = UIFont.systemFont(ofSize: 12)
+                } else {
+                    cell.saveButton.setTitle("Okuma Listesine Ekle", for: .normal)
+                    cell.saveButton.setTitleColor(.gray, for: .normal)
+                    cell.saveButton.titleLabel?.font = UIFont.systemFont(ofSize: 12)
+                }
+            }
+            return cell
+        }
+        return UICollectionViewCell()
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return collectionView == sliderCollectionView ? CGSize(width: collectionView.frame.width, height: collectionView.frame.height) : CGSize(width: collectionView.frame.width, height: collectionView.frame.height / 2 + 100)
```

Comment:
burayı daha okunabilir yapabilirsin.
conditionları tanımlarsın mesela sonra yerleştirirsin.

```suggestion
       let firstCondition = CGSize(width: collectionView.frame.width, height: collectionView.frame.height)
       let secondCond =  CGSize(width: collectionView.frame.width, height: collectionView.frame.height / 2 + 100)
        return collectionView == sliderCollectionView ? firstCondition : secondCond
```